datastore/db/bytebase/FertiScan/update_inspection_function.sql

```diff
@@ -0,0 +1,541 @@
+    SELECT id INTO location_id
+    FROM "fertiscan_0.0.11".location
+    WHERE address = input_org_info->>'address'
+    LIMIT 1;
+
+    -- Use upsert_location to insert or update the location
+    location_id := "fertiscan_0.0.11".upsert_location(location_id, input_org_info->>'address');
+
+    -- Extract the organization info ID from the input JSON or generate a new UUID if not provided
+    organization_info_id := COALESCE(NULLIF(input_org_info->>'id', '')::uuid, uuid_generate_v4());
+
+    -- Upsert organization information
+    INSERT INTO "fertiscan_0.0.11".organization_information (id, name, website, phone_number, location_id)
+    VALUES (
+        organization_info_id,
+        input_org_info->>'name',
+        input_org_info->>'website',
+        input_org_info->>'phone_number',
+        location_id
+    )
+    ON CONFLICT (id) DO UPDATE
+    SET name = EXCLUDED.name,
+        website = EXCLUDED.website,
+        phone_number = EXCLUDED.phone_number,
+        location_id = EXCLUDED.location_id
+    RETURNING id INTO organization_info_id;
+
+    RETURN organization_info_id;
+END;
+$$ LANGUAGE plpgsql;
+
+
+-- Function to upsert label information
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.11".upsert_label_information(
+    input_label jsonb,
+    company_info_id uuid,
+    manufacturer_info_id uuid
+)
+RETURNS uuid AS $$
+DECLARE
+    label_id uuid;
+BEGIN
+    -- Extract or generate the label ID
+    label_id := COALESCE(NULLIF(input_label->>'id', '')::uuid, uuid_generate_v4());
+
+    -- Upsert label information
+    INSERT INTO "fertiscan_0.0.11".label_information (
+        id, lot_number, npk, registration_number, n, p, k, company_info_id, manufacturer_info_id
+    )
+    VALUES (
+        label_id,
+        input_label->>'lot_number',
+        input_label->>'npk',
+        input_label->>'registration_number',
+        (NULLIF(input_label->>'n', '')::float),
+        (NULLIF(input_label->>'p', '')::float),
+        (NULLIF(input_label->>'k', '')::float),
+        company_info_id,
+        manufacturer_info_id
+    )
+    ON CONFLICT (id) DO UPDATE
+    SET lot_number = EXCLUDED.lot_number,
+        npk = EXCLUDED.npk,
+        registration_number = EXCLUDED.registration_number,
+        n = EXCLUDED.n,
+        p = EXCLUDED.p,
+        k = EXCLUDED.k,
+        company_info_id = EXCLUDED.company_info_id,
+        manufacturer_info_id = EXCLUDED.manufacturer_info_id
+    RETURNING id INTO label_id;
+
+    RETURN label_id;
+END;
+$$ LANGUAGE plpgsql;
+
+
+-- Function to update metrics: delete old and insert new
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.11".update_metrics(
+    p_label_id uuid,
+    metrics jsonb
+)
+RETURNS void AS $$
+DECLARE
+    metric_record jsonb;
+    metric_type text;
+    metric_value float;
+    metric_unit text;
+    unit_id uuid;
+BEGIN
+    -- Delete existing metrics for the given label_id
+    DELETE FROM "fertiscan_0.0.11".metric WHERE label_id = p_label_id;
```

Comment:
why are we deleting the metrics?

Reply:
As discussed, this is not ideal. We'll later implement a robust history system to allow for analytics.